### PR TITLE
Yanling update api doc

### PR DESCRIPTION
**Pull request by YANLING-H**
### Design Decisions
1. HTTP code: ([reference](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design?ranMID=46131&ranEAID=a1LgFw09t88&ranSiteID=a1LgFw09t88-P2cPlKaVA6qKTF.VPyGwMQ&epi=a1LgFw09t88-P2cPlKaVA6qKTF.VPyGwMQ&irgwc=1&OCID=AID2200057_aff_7806_1243925&tduid=(ir__ltpyxfcwtgkfb1vhjvm6l2w2iv2xvfanemq1v2dx00)(7806)(1243925)(a1LgFw09t88-P2cPlKaVA6qKTF.VPyGwMQ)()&irclickid=_ltpyxfcwtgkfb1vhjvm6l2w2iv2xvfanemq1v2dx00))
   1. If the resource cannot be found, the method should return 404 (Not Found)
   2. If a POST method creates a new resource, it returns the HTTP status code 201 (Created). 
 2. Use a “singular” name to denote the document resource archetype and the “plural” name to denote the collection resource archetype. ([reference](https://restfulapi.net/resource-naming/))
 3. Use dynamic URL parameters when getting a specific resource and use query parameters to filter resources

### CODE REVIEW

File: doc/API doc.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ user logs in
 
 ```
 { "code": 400, "errMessage": "the code is not sent successfully" }
-{ "code": 400, "errMessage": "the phone number doesn't exist" }
+{ "code": 404, "errMessage": "the phone number doesn't exist" }
```
Comment (YANLING-H) [inline doc/API doc.md:48]: Design decision 1

File: doc/API doc.md [hunk 2]
```diff
@@ -54,10 +54,9 @@ user logs in
 
 update user information on the profile page
 
-**URL**: `/api/user/update-profile`
+**URL**: `/api/user/:id/profile`
```
Comment (YANLING-H) [inline doc/API doc.md:58]: Design decision 3

File: doc/API doc.md [hunk 3]
```diff
@@ -89,14 +88,14 @@ collect user information for the new user
...
 | nickName    | String |                | `"Mike Chen"`     |
-| avatar      | String | save as base64 | `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABSgAAADoCAYAAAD..."`              |
+| avatar      | String | save as base64 | `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABSgAAADoCAYAAAD..."`   or empty string if user didn't upload the file           |
 
 **Success Response**:
 
 ```
-{ "code": 200,  "states": "success"}
+{ "code": 201,  "states": "success"}
 ```
```
Comment (YANLING-H) [inline doc/API doc.md:99]: Design decision 1

File: doc/API doc.md [hunk 4]
```diff
@@ -112,7 +111,7 @@ collect user information for the new user
 
 load user information for the profile page
 
-**URL**: `/api/user/info`
+**URL**: `/api/user/:id/profile`
```
Comment (YANLING-H) [inline doc/API doc.md:115]: Design decision 3

File: doc/API doc.md [hunk 5]
```diff
@@ -145,9 +144,9 @@ load user information for the profile page
 
 load all events to display on [this page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=19%3A222)
 
-**URL**: `/api/event/load-all`
+**URL**: `/api/events/load-all`
```
Comment (YANLING-H) [inline doc/API doc.md:148]: Design decision 2

File: doc/API doc.md [hunk 6]
```diff
@@ -217,24 +216,23 @@ add info about a new event
 **Success Response**:
 
 ```
-{ "code": 200,  "statue": "success"}
+{ "code": 201,  "statue": "success"}
 ```
```
Comment (YANLING-H) [inline doc/API doc.md:220]: Design decision 1

File: doc/API doc.md [hunk 7]
```diff
@@ -217,24 +216,23 @@ add info about a new event
...
 
 **Error Response**:
 
 ```
-{ "code":400, "errMessage": "host does not exist" }
+{ "code":404, "errMessage": "host does not exist" }
 // add 401
```
Comment (YANLING-H) [inline doc/API doc.md:226]: Design decision 1

File: doc/API doc.md [hunk 8]
```diff
@@ -217,24 +216,23 @@ add info about a new event
...
 
 ### [Get] Load Event by EventId
 
 **Description**:
 
-
 load event information for "event detail" pages: [page1](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=29%3A211), [page2](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=27%3A118), [page3](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=55%3A255)
 
-**URL**: `/api/event/id`
+**URL**: `/api/event/:id`
```
Comment (YANLING-H) [inline doc/API doc.md:236]: Design decision 3

File: doc/API doc.md [hunk 9]
```diff
@@ -281,7 +279,7 @@ load event information for "event detail" pages: [page1](https://www.figma.com/f
 **Error Response**:
 
 ```
-{ "code": 400, "errMessage": "event does not exist" }
+{ "code": 404, "errMessage": "event does not exist" }
 ```
```
Comment (YANLING-H) [inline doc/API doc.md:283]: Design decision 1

File: doc/API doc.md [hunk 10]
```diff
@@ -308,8 +306,8 @@ register event for the given user
 **Error Response**:
 
 ```
-{ "code": 400, "errMessage": "user does not exist"}
-{ "code": 400, "errMessage": "event does not exist"}
+{ "code": 404, "errMessage": "user does not exist"}
+{ "code": 404, "errMessage": "event does not exist"}
 { "code": 400, "errMessage": "user already registered this event"}
```
Comment (YANLING-H) [inline doc/API doc.md:311]: Design decision 1

File: doc/API doc.md [hunk 11]
```diff
@@ -319,7 +317,7 @@ register event for the given user
 
 load info for [page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=27%3A120) and [page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=113%3A168)
 
-**URL**: `/api/event/participate-event`
+**URL**: `/api/events/:userid/participated-participating-events`
```
Comment (YANLING-H) [inline doc/API doc.md:321]: Design decision 3

File: doc/API doc.md [hunk 12]
```diff
@@ -349,7 +347,7 @@ load info for [page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?no
 
 load events for [page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=113%3A354) and [page](https://www.figma.com/file/VwMPTLWxxXwuAr1E2twT0E/MeetIt?node-id=72%3A415)
 
-**URL**: `/api/event/host-events`
+**URL**: `/api/events/:userid/hosted-hosting-events`
```
Comment (YANLING-H) [inline doc/API doc.md:351]: Design decision 3

File: doc/API doc.md [hunk 13]
```diff
@@ -434,7 +432,7 @@ unenroll event
 
 Update event information
 
-**URL**: `/api/event/edit`
+**URL**: `/api/event/:id/update`
```
Comment (YANLING-H) [inline doc/API doc.md:436]: Design decision 3

File: doc/API doc.md [hunk 14]
```diff
@@ -468,7 +466,7 @@ Update event information
 
 find desired events according to filters
 
-**URL**: `/api/event/filter`
+**URL**: `/api/events/filter`
```
Comment (YANLING-H) [inline doc/API doc.md:470]: Design decision 2

**Review by yuliwilliam — APPROVED** [review]

**Pull request merged**